src/ng/rootScope.js

```diff
@@ -1375,6 +1374,7 @@ function $RootScopeProvider() {
 
         if (current.$$listenerCount[name] === 0) {
           delete current.$$listenerCount[name];
+          delete current.$$listeners[name];
```

Comment:
One issue brought up is that `$$listeners` and `$$listenerCount` often do not have the same lifetime. `$$listenerCount` will reach `0` only when this scope and all child scopes have no listeners.  I think that might convince me that this PR is no good as-is. But I still thinking cleaning up `$$listeners` when empty might be a decent idea. TBD...